src/python/pants/testutil/pants_integration_test.py

```diff
@@ -101,6 +117,189 @@ def join(
+            "RUN_PANTS_FROM_PEX",
+        ):
+            value = os.getenv(h)
+            if value is not None:
+                env[h] = value
+        hermetic_env = os.getenv("HERMETIC_ENV")
+        if hermetic_env:
+            for h in hermetic_env.strip(",").split(","):
+                value = os.getenv(h)
+                if value is not None:
+                    env[h] = value
+    else:
+        env = os.environ.copy()
+    if extra_env:
+        env.update(extra_env)
+    env.update(PYTHONPATH=os.pathsep.join(sys.path))
+
+    # Pants command that was called from the test shouldn't have a parent.
+    if "PANTS_PARENT_BUILD_ID" in env:
+        del env["PANTS_PARENT_BUILD_ID"]
+
+    return PantsJoinHandle(
+        command=pants_command,
+        process=subprocess.Popen(
+            pants_command,
+            env=env,
+            stdin=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            **kwargs,
+        ),
+        workdir=workdir,
+    )
+
+
+def run_pants_with_workdir(
+    command: Command,
+    *,
+    workdir: str,
+    hermetic: bool = True,
+    use_pantsd: bool = True,
+    config: Optional[Mapping] = None,
+    stdin_data: Optional[Union[bytes, str]] = None,
+    tee_output: bool = False,
+    **kwargs: Any,
+) -> PantsResult:
+    if config:
+        kwargs["config"] = config
+    handle = run_pants_with_workdir_without_waiting(
+        command, workdir=workdir, hermetic=hermetic, use_pantsd=use_pantsd, **kwargs
+    )
+    return handle.join(stdin_data=stdin_data, tee_output=tee_output)
+
+
+def run_pants(
+    command: Command,
+    *,
+    hermetic: bool = True,
+    use_pantsd: bool = True,
+    config: Optional[Mapping] = None,
+    extra_env: Optional[Mapping[str, str]] = None,
+    stdin_data: Optional[Union[bytes, str]] = None,
+    **kwargs: Any,
+) -> PantsResult:
+    """Runs Pants in a subprocess.
+
+    :param command: A list of command line arguments coming after `./pants`.
+    :param hermetic: If hermetic, your actual `pants.toml` will not be used.
+    :param use_pantsd: If True, the Pants process will use pantsd.
+    :param config: Optional data for a generated TOML file. A map of <section-name> ->
+        map of key -> value.
+    :param kwargs: Extra keyword args to pass to `subprocess.Popen`.
+    """
+    with temporary_workdir() as workdir:
+        return run_pants_with_workdir(
+            command,
+            workdir=workdir,
+            hermetic=hermetic,
+            use_pantsd=use_pantsd,
+            config=config,
+            stdin_data=stdin_data,
+            extra_env=extra_env,
+            **kwargs,
+        )
+
+
+# -----------------------------------------------------------------------------------------------
+# Environment setup.
+# -----------------------------------------------------------------------------------------------
+
+
+@contextmanager
+def setup_tmpdir(files: Mapping[str, str]) -> Iterator[str]:
```

Comment:
This feels like it should be a fixture. 

For example, the fixture could take a marker that provides the files, but if that marker isn't provided it could default to taking the value of `SOURCES` in the requesting module, or something like that. 

See, e.g., https://docs.pytest.org/en/stable/fixture.html#using-markers-to-pass-data-to-fixtures and https://docs.pytest.org/en/stable/fixture.html#fixtures-can-introspect-the-requesting-test-context.